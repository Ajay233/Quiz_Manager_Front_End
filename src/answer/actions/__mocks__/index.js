export const getAnswers = jest.fn()

export const setCurrentAnswer = jest.fn()

export const deleteAnswer = jest.fn()

export const addAnswer = jest.fn(() => { return { type: "ADD_ANSWER", payload: {} } })

export const updateAnswer = jest.fn()
