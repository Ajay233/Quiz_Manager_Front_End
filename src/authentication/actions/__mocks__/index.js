export const setUser = jest.fn()

export const setVerficationProcess = jest.fn()

export const setRedirect = jest.fn()

export const logOut = jest.fn()
