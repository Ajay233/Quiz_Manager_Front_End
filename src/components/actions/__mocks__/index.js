export const setLoaderState = jest.fn(() => { return { type: "SET_LOADER_STATE", payload: {} } })
