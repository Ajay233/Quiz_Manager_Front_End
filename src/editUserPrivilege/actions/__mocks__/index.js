export const setUserResults = jest.fn( () => { return { type: "SET_USER_RESULTS", payload: "" } })

export const clearUserResults = jest.fn()
