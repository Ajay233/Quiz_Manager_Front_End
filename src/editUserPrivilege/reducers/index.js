import { userResults } from '../factory/userResultsFactory';

export const setUserResultsReducer = (state=userResults, action) => {

  switch (action.type) {
    case "SET_USER_RESULTS": return action.payload;
    default: return state;
  }
}
