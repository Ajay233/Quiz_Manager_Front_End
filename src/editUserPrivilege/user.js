import React from 'react';

const User = (props) => {
  const { forename, surname, email, permission, verified } = props.userResults;
  return(
      <div>
        <div>{`Name: ${forename} ${surname}`}</div>
        <div>{`Email: ${email}`}</div>
        <div>{`Permission: ${permission}`}</div>
        <div>{`Verified: ${verified === "true" ? "Yes" : "No"}`}</div>
      </div>
    );
}

export default User
