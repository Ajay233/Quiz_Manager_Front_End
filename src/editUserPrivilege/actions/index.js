import { getUsingParams } from '../../axiosRequests/requests'
import { setNotification } from '../../notifications/actions'
import { sessionExpired } from '../../utils/session'

export const setUserResults = (param, jwt) => {
  return (dispatch) => {
    return getUsingParams("users/findByEmail", param, jwt).then((response) => {
      dispatch({
        type: "SET_USER_RESULTS",
        payload: response.data
      })
    }).catch((error) => {
      if(error.response.status === 403){
        sessionExpired(dispatch);
      } else {
        const msg = error.response.data;
        dispatch(setNotification(msg, "error", true));
        console.log(error.response)
      }
    });
  }
}

export const clearUserResults = () => {
  return {
    type: "CLEAR_USER_RESULTS",
    payload: {}
  }
}
