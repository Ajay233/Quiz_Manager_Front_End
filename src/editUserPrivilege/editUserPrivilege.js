import React from 'react'
import { connect } from 'react-redux'
import FindUserByEmail from '../forms/findUserByEmail'
import Notification from '../notifications/notifications'
import User from './user'
import UpdatePermission from '../forms/updatePermission'

import { setUserResults, clearUserResults } from './actions'
import { setNotification } from '../notifications/actions'

import '../stylesheets/editUserPrivilege.css'

export class EditUserPrivilege extends React.Component {

  componentWillUnmount(){
    this.props.clearUserResults()
  }

  renderUser = () => {
    return Object.entries(this.props.userResults).length === 0 ? null : <div className="centeredContent"><hr/><User userResults={this.props.userResults}/></div>;
  }

  renderUpdatePermission = () => {
    const {userResults, userData, setNotification, clearUserResults} = this.props
    return Object.entries(this.props.userResults).length === 0 ? null : <UpdatePermission userResults={userResults} clearUserResults={clearUserResults} userData={userData} setNotification={setNotification} />;
  }

  render(){
    const { userData, setUserResults, setNotification } = this.props;
    return(
      <div id="editUserPrivilege">
        <Notification />
        <div id="editPrivilegeHeading">
          <img id="editPrivilegeImg" src={require("../public/icons/privilege.png")} alt=""/>
          <div id="editPrivilegeTitle" className="">Edit user privilege</div>
        </div>
        <div>
        <FindUserByEmail userData={userData} setUserResults={setUserResults} setNotification={setNotification} />
        </div>
        {this.renderUser()}
        {this.renderUpdatePermission()}

      </div>
    );
  }
}

export const mapStateToProps = (state) => {
  return {
    userData: state.userData,
    userResults: state.userResults
  }
}

export default connect(mapStateToProps, { setUserResults, clearUserResults, setNotification })(EditUserPrivilege)
