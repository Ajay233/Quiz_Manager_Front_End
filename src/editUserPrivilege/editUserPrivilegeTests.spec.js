import React from 'react'
import { EditUserPrivilege, mapStateToProps } from './editUserPrivilege'
import { Provider } from 'react-redux'
import { shallow, mount } from 'enzyme'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import history from '../history'
import { clearUserResults } from './actions/__mocks__/index'

// jest.mock("./actions")

const mockStore = configureStore({})

describe("User", () => {

  let store;
  let userData;
  let userResults;
  let notificationData;

  beforeEach(() => {
    store = mockStore({
      notificationData:{
        message: "",
        type: "",
        show: false,
        timed: true
      }
    })

    userData = {
      jwt: "jwt"
    }

    userResults = {
      forename: "testForename",
      surname: "testSurname",
      email: "testEmail",
      permission: "USER",
      verified: true
    }

    notificationData = {
      message: "",
      type: "",
      show: false,
      timed: true
    }
  })

  it("should render a user's details using the props passed to it", () => {

    const component = renderer.create(
      <Provider store={store}>
        <EditUserPrivilege
          userData={userData}
          userResults={userResults}
          notificationData={notificationData}
        />
      </Provider>
    )

    expect(component).toMatchSnapshot()

  })

  it("should not render the edit privilege form if the userResults is empty", () => {

    userResults = {}

    const component = renderer.create(
      <Provider store={store}>
        <EditUserPrivilege
          userData={userData}
          userResults={userResults}
          notificationData={notificationData}
        />
      </Provider>
    )

    expect(component).toMatchSnapshot()

  })

  it("should call clearUserResults on unmount", () => {

    const wrapper = mount(
      <Provider store={store}>
        <EditUserPrivilege
          userData={userData}
          clearUserResults={clearUserResults}
          userResults={userResults}
          notificationData={notificationData}
        />
      </Provider>
    )

    wrapper.unmount()

    expect(clearUserResults).toHaveBeenCalledTimes(1)

  })
})

describe("mapStateToProps", () => {
  it("should map state to props", () => {
    const appState = {
      userData: {
        jwt: "jwt"
      },
      userResults: {
        forename: "testForename",
        surname: "testSurname",
        email: "testEmail",
        permission: "USER",
        verified: true
      }
    }

    const componentState = mapStateToProps(appState)
    expect(componentState).toEqual(appState)
  })
})
