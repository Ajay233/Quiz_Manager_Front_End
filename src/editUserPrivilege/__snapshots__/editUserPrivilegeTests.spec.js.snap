// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User should not render the edit privilege form if the userResults is empty 1`] = `
<div
  id="editUserPrivilege"
>
  <div
    id="editPrivilegeHeading"
  >
    <img
      alt=""
      id="editPrivilegeImg"
      src="privilege.png"
    />
    <div
      className=""
      id="editPrivilegeTitle"
    >
      Edit user privilege
    </div>
  </div>
  <div>
    <div>
      <form
        className="form-centered"
        onSubmit={[Function]}
      >
        <div>
          <label>
            Email:
          </label>
          <input
            className="inputBox"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            value=""
          />
        </div>
        <button
          className="submit"
        >
          <i
            className="fas fa-search"
          />
           Search
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`User should render a user's details using the props passed to it 1`] = `
<div
  id="editUserPrivilege"
>
  <div
    id="editPrivilegeHeading"
  >
    <img
      alt=""
      id="editPrivilegeImg"
      src="privilege.png"
    />
    <div
      className=""
      id="editPrivilegeTitle"
    >
      Edit user privilege
    </div>
  </div>
  <div>
    <div>
      <form
        className="form-centered"
        onSubmit={[Function]}
      >
        <div>
          <label>
            Email:
          </label>
          <input
            className="inputBox"
            name="email"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            value=""
          />
        </div>
        <button
          className="submit"
        >
          <i
            className="fas fa-search"
          />
           Search
        </button>
      </form>
    </div>
  </div>
  <div
    className="centeredContent"
  >
    <hr />
    <div>
      <div>
        Name: testForename testSurname
      </div>
      <div>
        Email: testEmail
      </div>
      <div>
        Permission: USER
      </div>
      <div>
        Verified: No
      </div>
    </div>
  </div>
  <div>
    <form
      className="form-centered"
      onSubmit={[Function]}
    >
      <div>
        <label>
          Set Permission:
        </label>
        <div>
          <select
            className="inputBox select-medium"
            name="permission"
            onBlur={[Function]}
            onChange={[Function]}
            onDragStart={[Function]}
            onDrop={[Function]}
            onFocus={[Function]}
            value=""
          >
            <option>
              Select permission level
            </option>
            <option
              value="USER"
            >
              USER
            </option>
            <option
              value="READ-ONLY"
            >
              READ-ONLY
            </option>
            <option
              value="ADMIN"
            >
              ADMIN
            </option>
            <option
              value="SUPER-USER"
            >
              SUPER-USER
            </option>
          </select>
        </div>
      </div>
      <button
        className="submit"
      >
        <i
          className="far fa-save"
        />
         Save
      </button>
    </form>
  </div>
</div>
`;
